requirements.txt

```diff
@@ -1,8 +1,8 @@
+jax[cpu] >= 0.3.2, < 0.5.0
```

Comment:
minor point but can we keep this in alphabetical order?

Also don't we need to bump the minimum jax version?

Reply:
Yes, we need to bump it up. Are we ok with `0.4.24`?

Reply:
I changed this to make `test_jax` work better. In the alphabetical order (when jax comes later) pip was having trouble finding proper versions. 

I made some other changes after that, maybe this is not an issue anymore but basically this prioritizes `jax` version over others in a way

Reply:
What are the issues with older versions of jax?

Reply:
- `v0.4.23`-`v0.4.13` fails due to some scipy function
![image](https://github.com/user-attachments/assets/5d6b540f-e515-4526-9948-c3f4e57e0562)
I don't know why pip doesn't choose proper scipy version, but it is probably due to `jax` requirements which I cannot fix.

- `v0.4.12` and older won't even past the pip install phase.

Reply:
for the scipy one, if you manually then install the correct version of scipy does everything work? If so we can probably leave our jax requirement at 0.4.13, assuming that if people have an old version of jax installed then they also have an old version of scipy which should be compatible.

Reply:
I didn't try that. Let me check

Reply:
also I think we can remove the `[cpu]` flag for jax, by default it installs the cpu version and this would make it less annoying for installing gpu stuff

Reply:
if we remove `[cpu]` than we need to add `jaxlib` to the requirements

Reply:
I think all the recent versions install jaxlib automatically

Reply:
btw, even if I install older `scipy` version, I get new errors
![image](https://github.com/user-attachments/assets/8545c685-0aeb-4d75-b674-ad8364b3bc30)
I just checked 0.4.13 locally because I need to add some logic to CI to test it. If you wanna test, feel free to do it here